### PR TITLE
ENT-3759: Test on GitHub Actions

### PR DESCRIPTION
**Pull request by m-horky**
* Card ID: ENT-3759

### CODE REVIEW

**Review by ptoscano — CHANGES_REQUESTED** [review]
Looks like a good start!

I left some minor notes & questions here and there, nothing too problematic IMHO.

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  pytest:
+    name: "pytest"
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "CentOS Stream 9"
+            image: "quay.io/centos/centos:stream9"
+            pytest_args: '--deselect test/rhsmlib/facts/test_hwprobe.py::HardwareProbeTest::test_networkinfo --deselect test/rhsmlib_test/test_facts.py::TestFactsDBusObject::test_GetFacts'
```
[the remark this answers is not in the document]
Comment (jirihnidek) [reply .github/workflows/pytest.yml:17]: This class and method is located in the following file: `test/rhsmlib/dbus/test_facts.py`. If there is good reason to not run following unit tests, then this reason should be shared in the comment of the code.

**Review by jirihnidek — CHANGES_REQUESTED** [review]
I like it! :+1: I have few comments notes and suggestions too.

File: .github/workflows/pytest.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ptoscano) [reply .github/workflows/pytest.yml]: though, if it worked in the first revision, then most likely the exclusion of `test_GetFacts` is not needed?

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            # The 'test_networkinfo' breaks in CentOS container because it has IPv6 disabled.
```
[the remark this answers is not in the document]
Comment (ptoscano) [reply .github/workflows/pytest.yml:18]: what I meant is something different:
https://github.com/candlepin/subscription-manager/blob/e9b945a1bb79a413a5271a2f74ed5d69848873ca/test/rhsmlib/facts/test_hwprobe.py#L581-L585
ie `test_networkinfo` requires that both `network.ipv4_address` and `network.ipv6_address` facts are available, which may not be the case if either is disabled

File: test/rhsmlib/dbus/test_facts.py [hunk 3]
```diff
@@ -29,8 +29,18 @@ def setUpClass(cls) -> None:
             name="get_virt_info",
         )
         cls.patches["get_virt_info"] = get_virt_info_patch.start()
+        cls.patches["get_virt_info"].return_value = {}
```
[the remark this answers is not in the document]
Comment (ptoscano) [reply test/rhsmlib/dbus/test_facts.py]: ah sure, separate PR is what I had in mind -- waiting for it ;)

File: .github/workflows/pytest.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+      - name: "Run container-pre-test.sh"
+        run: |
+          bash scripts/container-pre-test.sh
+      - name: "Run pytest"
+        env:
+          PYTEST_ADDOPTS: "--color=yes --code-highlight=yes --showlocals"
+        run: |
+          SUBMAN_TEST_IN_CONTAINER=1 dbus-run-session \
+            python3 -m pytest ${{ matrix.pytest_args }}
```
[the remark this answers is not in the document]
Comment (ptoscano) [reply .github/workflows/pytest.yml]: oh nice action, looking forward to it :+1:

**Review by ptoscano — APPROVED** [review]
LGTM, thanks!

I'll let Jirka review this as well, as he had notes previously.

**Review by jirihnidek — APPROVED** [review]
ACK

**Pull request merged**